### DISCUSSION TITLE
Precompiling Assets When Building

### DISCUSSION THREAD

**Discussion by partydrone** [nickjj/docker-rails-example#44] [Q&A]
I'm using your Dockerfile as a starting point for a Rails 5 project. I'm running into problems when the build gets to the assets:precompile step—it's trying to connect to the database and fails. I tried setting `config.assets.initialize_on_precompile = false`, but that didn't resolve the issue.

When the database host is missing, or set to "localhost," Rails tries connecting to the database via socket. If I add the host as part of the precompile command (`MYSQL_HOST=db SECRET_KEY_BASE=dummyvalue rails assets:precompile`), then it doesn't like the host ("Unknown MySQL server host 'db').

Have you ever run into this issue on your projects using this Dockerfile?

**Comment by nickjj** [comment]
Hi,

I vaguely remember this being an issue with Rails 5.

I'm almost sure I was able to set `DATABASE_URL=postgresql://dummy:dummy@127.0.0.1/dummy SECRET_KEY_BASE=dummy rails assets:precompile` and it worked. In your case try swapping `postgresql` with `mysql`. I remember having to set the whole DB URI for it to work.

**Reply by partydrone** [reply]
It's a mystery:

```
------
 > [assets 10/10] RUN if [ "production" != "development" ]; then   DATABASE_URL=mysql2://dummy:dummy@127.0.0.1/dummy SECRET_KEY_BASE=dummyvalue rails assets:precompile; fi:
#0 4.981 sh: 1: cd: can't cd to /src/.git
#0 5.520 rails aborted!
#0 5.520 Mysql2::Error: Can't connect to MySQL server on '127.0.0.1' (111 "Connection refused")
```

**Reply by partydrone** [reply]
Even if I set `config.assets.initialize_on_precompile = false` in application.rb.

**Reply by nickjj** [reply]
I never tried this but using https://github.com/nulldb/nulldb would work, here's how to pull it together https://stackoverflow.com/a/46258605.

I don't love it since it involves installing a third party gem, but if you have to use Rails 5 it'll get the job done.

**Reply by partydrone** [reply]
I did come across that in my research. I'll give it a try. Another approach is to remove all initializers, then add them back, one-by-one to find the culprit(s), and wrap them in a conditional:

```ruby
unless defined?(::Rake::SprocketsTask)
  # Initializer code here …
end
```

This may also be necessary for the routes file. 😩

**Reply by nickjj** [reply]
Yeah, it sounds like you're getting into pretty murky territory with a bunch of customizations.

After thinking it a bit more, the null adapter approach will solve it without having to disable anything and then if you upgrade to a newer version of Rails you can drop the dependency since it doesn't connect to the DB anymore. It feels better when you classify it as a temporary dependency, almost like a polyfill.
